### ISSUE TITLE
SCIM Parser throwing exception on valid filter

### ISSUE THREAD

**Issue by nikhilbadgujar** [simpleidserver/SimpleIdServer#419]
**Issue description:** 

**Setup:**
- I am using https://www.nuget.org/packages/SimpleIdServer.Scim.Parser/2.0.25 package to convert filter string into expression.
```
   using SimpleIdServer.Scim.Parser.Expressions;
   var filter = "(id eq user1 or id eq user2) or ((id eq user3 or id eq user4) or (id eq user5 or id eq user6))";
   SCIMFilterParser.Parse(filter);
```

**Expected:**
- Expression tree based on the filter

**Actual:**
- System.NullReferenceException is thrown with message `{"Object reference not set to an instance of an object."}`
- Stack trace- 
```
   at SimpleIdServer.Scim.Parser.Expressions.SCIMAttributeExpression.ProtectedClone()
   at SimpleIdServer.Scim.Parser.Expressions.SCIMComparisonExpression.Clone()
   at SimpleIdServer.Scim.Parser.Expressions.SCIMLogicalExpression.Clone()
   at SimpleIdServer.Scim.Parser.Expressions.SCIMLogicalExpression.Clone()
   at SimpleIdServer.Scim.Parser.SCIMFilterParser.TransformStringFiltersIntoSCIMLogicalExpression(IEnumerable`1 filters)
```

I am able to get SCIMExpression for filter = `(id eq user1 or id eq user2) or (id eq user3 or id eq user4)`, but when I try to increase the OR operators then it throws error.
Can someone please take a look at this issue ?
Let me know if any more details are needed on this issue.

Thank you

**Comment by simpleidserver** [comment]
Hi,

Indeed there is an issue in the library. The issue is fixed in the branch `release/2.1.0`.

KR,

SID

**Comment by simpleidserver** [comment]
Hello,

The Nuget Package version 3.0.0 is available, it contains the modifications made on the branch `release\2.1.0`.

KR,

SID
